### PR TITLE
samples: peripheral: lpuart: Reduce power consumption for nrf9160dk

### PR DESCRIPTION
**Pull request by nordic-krch**
On nrf9160dk >=0.15.0 it is required to initilize the modem to get
lowest possible power consumption. Converting low power uart sample
to use nrf9160dk_nrf9160ns target with modem library enabled.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

File: samples/peripheral/lpuart/boards/nrf9160dk_nrf9160ns.conf [hunk 1]
```diff
@@ -9,3 +9,9 @@ CONFIG_UART_1_INTERRUPT_DRIVEN=n
 CONFIG_UART_1_NRF_HW_ASYNC=y
 CONFIG_UART_1_NRF_HW_ASYNC_TIMER=2
 CONFIG_NRFX_TIMER2=y
+
+# Setup modem to get lowest power consumption
+CONFIG_NRF_MODEM_LIB=y
+CONFIG_NETWORKING=y
+CONFIG_NET_SOCKETS=y
+CONFIG_HEAP_MEM_POOL_SIZE=32768
```
Comment (lemrey) [inline samples/peripheral/lpuart/boards/nrf9160dk_nrf9160ns.conf:17]: This is leftover?

Comment (nordic-krch) [reply samples/peripheral/lpuart/boards/nrf9160dk_nrf9160ns.conf:17]: this, only this is needed

**Review by lemrey — APPROVED** [review]
Thanks

File: samples/peripheral/lpuart/boards/nrf9160dk_nrf9160ns.overlay [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /* SPDX-License-Identifier: LicenseRef-Nordic-5-Clause */
 
 &uart1 {
+	status = "ok";
```
Comment (anangl) [inline samples/peripheral/lpuart/boards/nrf9160dk_nrf9160ns.overlay:4]:
```suggestion
	status = "okay";
```
The "ok" form is deprecated.

Comment (nordic-krch) [reply samples/peripheral/lpuart/boards/nrf9160dk_nrf9160ns.overlay]: fixed

File: samples/peripheral/lpuart/sample.yaml [hunk 3]
```diff
@@ -5,14 +5,14 @@ tests:
   samples.peripheral.lpuart:
     build_only: true
     build_on_all: true
-    platform_allow: nrf52dk_nrf52832 nrf52833dk_nrf52833 nrf52840dk_nrf52840 nrf9160dk_nrf9160
+    platform_allow: nrf52dk_nrf52832 nrf52833dk_nrf52833 nrf52840dk_nrf52840 nrf9160dk_nrf9160ns
```
Comment (anangl) [inline samples/peripheral/lpuart/sample.yaml:8]: Please update also documentation, it still says that the target for nRF9160 DK is `nrf9160dk_nrf9160`: https://github.com/nrfconnect/sdk-nrf/blob/e3716141c15d37f743bee03a025a2b5505288c32/samples/peripheral/lpuart/README.rst#L25

**Review by anangl — CHANGES_REQUESTED** [review]

**Pull request closed without merging**